### ISSUE TITLE
Case timeline

### ISSUE THREAD

**Issue by laurjpeterson** [department-of-veterans-affairs/caseflow#5592]
## Background
When Board users view data about an appeal, understanding the "story" of an appeal, or where the appeal has been in the Board and in what sequence, is important context to help them complete their tasks. 

We hypothesize that early Queue users, attorneys and judges, will need to see this information when viewing appeal data in order to draft decisions. We also hypothesize that this information is useful for any Board user, especially those who do not have appeals signed to them. 

In this issue, we are documenting requirements for what should be included in the appeal timeline section of Queue. The goal of this section is to give the Board user context on where in the appeals process the appeal is, and what notable actions/tasks/changes have been made.

## Goals
- Give any Board user context on where in the appeals process the appeal is
- Highlight important appeals process actions/tasks/changes have been made along the story of the appeal

**Internal goals**
- Reuse appeals status logic wherever possible
- Distance ourselves from the concept of VACOLS locations - avoid use of the prior locations table for timelines for legacy appeals

**Stretch goal?**
- Give a Board user context on all Veterans' appeals and how their timelines interact, where applicable

## Progress made so far
- Initial timeline for RAMP appeals pilot (October 2018)
- Reworking how tasks should be named/appear to users - https://github.com/department-of-veterans-affairs/caseflow/issues/7508
- Introduced canceled icons for canceled tasks and withdrawn appeals 

### User Stories
- As a Board employee, I want to see the pertinent procedural history of an appeal, so that I know how it needs to be processed, routed, interpreted. 
  - E.g. It had a hearing, but transcript hasn't been uploaded yet
- As a Board employee, I want to see the pertinent procedural history of an appeal and it's current status, so I can answer inquiries (Congressional, external, internal, etc.)

### Epic Mural
- This [Mural](https://app.mural.co/t/workqueue2001/m/workqueue2001/1531849576358/38b753bff1e25963a76d8ebecf9679d6445c39b3) continues this epic description.

#### Legacy
- Rating decision date(s)
- NOD date
- SOC(s) date
- Form 9 date
- SSOC date
- Remand Return date
- Certification date
- Activation date
- Assigned to OMO date
- Back from OMO date (reassigned to Judge)
- Merged appeals?

#### AMA
- Supplemental claim 
- Higher level review
- Waiting for evidence submission? for 90 day evidence submission window for 2 of the 3 board lanes

#### Both
- Rating decision date(s)
- Assignment to VSO
   - IHP
- Waiting for hearing to be scheduled
   - Schedule hearing
- Hearing scheduled
   - Hold hearing
- Hearing held date + hearing disposition
- Assignment to translation
   - Translate documents
- Assignment to transcription
   - Transcribe hearing
- Waiting assignment to judge
- Assignment to judge
   - Review decision
   - Sign decision
- Assignment to attorney
   - Draft decision
- Assignment to admin / hold placed (see below)
  - FOIA request
  - IHP 
  - Extension request
- Assignment to BVA dispatch
  - Outcode decision
- BVA decision date
- CAVC remand date
- CAVC decision date
- Motions (e.g. AOD)

## Open Questions
- How should we emphasize blocking steps in this timeline (e.g. FOIA)
- How granular should these steps go? e.g. at transcription vs. at contractor, waiting for transcript to be returned
- What mail should we prioritize for this timeline?

**Comment by laurjpeterson** [comment]
From Appeals Status

## Attributes
- [x] `updated`
- [x] `type`
- [x] `active`
- [x] `incompleteHistory`
- [x] `aod`
- [x] `location`
- [x] `status`
- [x] `events`
- [x] `alerts`
- [x] `programArea`
- [x] `aoj`
- [x] `issues`
 - [x] `active`
 - [x] `last_action` + `date`
 - [x] `description`
 - [x] `diagnosticCode`
- [ ] `description` _(current sprint)_
- [ ] `docket` _(current sprint)_
 - [ ] `total` + `ahead` _(current sprint)_
 - [ ] `ready` _(current sprint)_
 - [ ] `front` _(current sprint)_
 - [ ] `eta`
- [ ] `evidence`

## Statuses
- [x] `scheduled_hearing`
 - [ ] `details`
- [x] `pending_hearing_scheduling`
 - [ ] `details`
- [x] `on_docket`
 - [ ] `details`
- [x] `pending_certification_ssoc`
 - [ ] `details`
- [x] `pending_certification`
 - [ ] `details`
- [x] `pending_form9`
 - [ ] `details`
- [x] `pending_soc`
 - [ ] `details`
- [x] `stayed`
 - [ ] `details`
- [x] `at_vso`
 - [ ] `details`
- [x] `bva_development`
 - [ ] `details`
- [x] `decision_in_progress`
 - [ ] `details`
- [x] `bva_decision`
 - [ ] `details`
- [x] `field_grant`
 - [ ] `details`
- [x] `withdrawn`
 - [ ] `details`
- [x] `ftr`
 - [ ] `details`
- [x] `ramp`
 - [ ] `details`
- [x] `death`
 - [ ] `details`
- [x] `reconsideration`
 - [ ] `details`
- [x] `other_close`
 - [ ] `details`
- [x] `remand_ssoc`
 - [ ] `details`
- [x] `remand`
 - [ ] `details`

## Events
- [x] `claim_decision`
- [x] `nod`
- [x] `soc`
- [x] `form9`
- [x] `ssoc`
- [x] `certified`
- [x] `remand_return`
- [x] `transcript`
- [x] `cavc_decision`
- [x] `ramp_notice`
- [x] `bva_decision`
- [x] `field_grant`
- [x] `withdrawn`
- [x] `ftr`
- [x] `ramp`
- [x] `death`
- [x] `merged`
- [x] `record_designation`
- [x] `reconsideration`
- [x] `vacated`
- [x] `other_close`
- [x] `hearing_held`
- [x] `hearing_cancelled`
- [x] `hearing_no_show`
- [ ] `dro_hearing_held`
- [ ] `dro_hearing_cancelled`
- [ ] `dro_hearing_no_show`

## Alerts
- [x] `form9_needed`
 - [ ] `details`
- [x] `scheduled_hearing`
 - [ ] `details`
- [x] `hearing_no_show`
 - [ ] `details`
- [x] `held_for_evidence`
 - [ ] `details`
- [x] `cavc_option`
 - [ ] `details`
- [x] `ramp_eligible`
 - [ ] `details`
- [x] `ramp_ineligible`
 - [ ] `details`
- [ ] `decision_soon` _(current sprint)_
 - [ ] `details`
- [ ] `blocked_by_vso` _(current sprint)_
 - [ ] `details`
- [ ] `scheduled_dro_hearing`
 - [ ] `details`
- [ ] `dro_hearing_no_show`
 - [ ] `details`

**Comment by laurjpeterson** [comment]
1. Who sees what?
2. Which types of events are only associated with a few users?

**Comment by laurjpeterson** [comment]
Took a first pass at a Case timeline epic [mural](https://app.mural.co/t/workqueue2001/m/workqueue2001/1531849576358/38b753bff1e25963a76d8ebecf9679d6445c39b3) for a legacy case. It attempts to show different data we could use - from appeals status to Caseflow. Lots of assumptions here, so this is still a work in progress to help us generate questions.

**Comment by mkhandekar** [comment]
Consider adding date of motion and user who made the motion for AOD in the timeline, from https://github.com/department-of-veterans-affairs/caseflow/issues/6257#issuecomment-421125993

> Today's date. We're operating on the assumption that when the user clicks "Submit", we capture today's date and log it in the background.

> Created by. Assuming we can capture this depending on the user that's logged into Caseflow and record it in the background.

**Comment by laurjpeterson** [comment]
@mdbenjam @mkhandekar @lowellrex @sneha-pai and @laurjpeterson discussed case timeline on 9/25/2018. we roughly went through (1) each of our ideas around what case timeline is (2) case timeline goals and users (3) what data is available to populate a case timeline and (4) some explorations of how to design and implement. 

#### Outcome of the discussion
- Our high level goal for case timeline: telling "the story of an appeal"
- Focus first on these goals: **who, what, and when**
   - Show dates - progress in time
   - Show ownership - who is assigned a given task
   - Show context in plain language - what is the task that is assigned
- Focus first on these data elements:
   - Task type
   - Date (thinking task completed date)
   - User assigned the task

@mkhandekar - can you add the photo of the whiteboard to this comment?

**Comment by mkhandekar** [comment]
Here's a picture of the discussion / whiteboard:

![img_0124](https://user-images.githubusercontent.com/1598889/46152170-731f6b80-c23e-11e8-8fb5-33e15171c61d.jpg)

**Comment by mkhandekar** [comment]
@sneha-pai and I walked through my sketches that tried to consolidate many of the ideas we talked about during this whiteboarding session.

https://app.mural.co/t/workqueue2001/m/workqueue2001/1531849576358/38b753bff1e25963a76d8ebecf9679d6445c39b3  (scroll to the right >>> )

Next steps are for @sneha-pai to: 
1.  walk this more-ideal state to a MVP version 2, a more lightweight and simple version that still satisfies the requirements @laurjpeterson noted in her note above. 
2. connect with @cmgiven / Appeals Status and trying to align (as much makes sense) the high level status' Veterans see with the more detailed internal status' internal VA folks see.

❤️ 👋
